samples/core/src/main/java/io/oasp/gastronomy/restaurant/general/common/impl/security/BaseUserDetailsService.java

```diff
@@ -0,0 +1,68 @@
+package io.oasp.gastronomy.restaurant.general.common.impl.security;
+
+import java.security.Principal;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+
+import io.oasp.gastronomy.restaurant.general.common.api.UserProfile;
+import io.oasp.gastronomy.restaurant.general.common.api.Usermanagement;
+import io.oasp.module.security.common.base.accesscontrol.AbstractAccessControlBasedAuthenticationProvider;
+
+/**
+ * This..
+ */
+@Named("BaseUserDetailsService")
+public class BaseUserDetailsService<U extends UserDetails, P extends Principal>
+    extends AbstractAccessControlBasedAuthenticationProvider implements UserDetailsService {
+
+  /** Logger instance. */
+  private static final Logger LOG = LoggerFactory.getLogger(BaseUserDetailsService.class);
+
+  private Usermanagement usermanagement;
+
+  /**
+   * @param usermanagement the {@link Usermanagement} to set
+   */
+  @Inject
+  public void setUsermanagement(Usermanagement usermanagement) {
+
+    this.usermanagement = usermanagement;
+  }
+
+  @Inject
+  private InMemoryUserDetailsManager inMemoryUserDetailsManager;
+
+  @Override
+  public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+
+    Set<GrantedAuthority> authorities = getAuthorities(username);
+
+    UserDetails user = this.inMemoryUserDetailsManager.loadUserByUsername(username);
```

Comment:
that was also an outdated version :( But I agree on your point. If for the sake of simplicity we want to use the in-memory authentication: username,password and role have to be hard-coded. An alternative would be to add a row in the database containing a (encrypted ?) password. This would also solve the need for two ProviderManagers (one for the password check and one for the permissions)